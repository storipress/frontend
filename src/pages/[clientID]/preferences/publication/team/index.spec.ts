import { expect, it } from 'vitest'
import { fireEvent, waitFor, within } from '@testing-library/vue'
import Team from './index.vue'
import { render } from '~/test-helpers'

it('render settings team', async () => {
  const { getByText, getAllByTestId } = render(Team)

  await waitFor(() => {
    expect(getByText('harvey contributor')).toBeVisible()
  })
  expect(getByText('ariel+01@storipress.com')).toBeVisible()

  expect(getAllByTestId('team-table-name-column')).not.toEqual([])
  expect(getAllByTestId('team-table-name-column')).toMatchSnapshot()
})

describe('sort settings team', () => {
  it('sort by name', async () => {
    const { getByText, getAllByTestId, getByRole } = render(Team)

    await waitFor(() => {
      expect(getByText('harvey contributor')).toBeVisible()
    })

    const sortNameButton = getByRole('columnheader', {
      name: /name/i,
    })
    await fireEvent.click(sortNameButton)
    expect(getAllByTestId('team-table-name-column')).toMatchSnapshot()
  })

  it('sort by status', async () => {
    const { getByText, getAllByTestId, getByRole } = render(Team)

    await waitFor(() => {
      expect(getByText('harvey contributor')).toBeVisible()
    })

    const sortNameButton = getByRole('columnheader', {
      name: /status/i,
    })
    await fireEvent.click(sortNameButton)
    expect(getAllByTestId('team-table-name-column')).toMatchSnapshot()
  })

  it('sort by role', async () => {
    const { getByText, getAllByTestId, getByRole } = render(Team)

    await waitFor(() => {
      expect(getByText('harvey contributor')).toBeVisible()
    })

    const sortNameButton = getByRole('columnheader', {
      name: /role/i,
    })
    await fireEvent.click(sortNameButton)
    expect(getAllByTestId('team-table-name-column')).toMatchSnapshot()
  })

  it('sort by desks', async () => {
    const { getByText, getAllByTestId, getByRole } = render(Team)

    await waitFor(() => {
      expect(getByText('harvey contributor')).toBeVisible()
    })

    const sortNameButton = getByRole('columnheader', {
      name: /desks/i,
    })
    await fireEvent.click(sortNameButton)
    expect(getAllByTestId('team-table-name-column')).toMatchSnapshot()
  })
})

describe('open dropdown', () => {
  it("user's status is active", async () => {
    const { getByText, getByRole } = render(Team)

    await waitFor(() => {
      expect(getByText('harvey contributor')).toBeVisible()
    })

    const row = getByRole('row', {
      name: /test testerson/i,
    })

    const dropdownButton = within(row).getByRole('button', {
      name: /open-dropdown/i,
    })

    await fireEvent.click(dropdownButton)
    expect(within(row).getAllByRole('menu')[0]).toBeVisible()
    expect(within(row).getByTestId('dropdown-menu')).not.toHaveTextContent('Unsuspend user')
    expect(within(row).getByTestId('dropdown-menu')).toHaveTextContent('Suspend user')
  })

  it("user's status is suspended", async () => {
    const { getByText, getByRole } = render(Team)

    await waitFor(() => {
      expect(getByText('harvey contributor')).toBeVisible()
    })

    const row = getByRole('row', {
      name: /storipress helper/i,
    })
    const dropdownButton = within(row).getByRole('button', {
      name: /open-dropdown/i,
    })

    await fireEvent.click(dropdownButton)
    expect(within(row).getAllByRole('menu')[0]).toBeVisible()
    expect(within(row).getByTestId('dropdown-menu')).not.toHaveTextContent('Suspend user')
    expect(within(row).getByTestId('dropdown-menu')).toHaveTextContent('Unsuspend user')
  })

  it("user's status is invited", async () => {
    const { getByText, getByRole } = render(Team)

    await waitFor(() => {
      expect(getByText('harvey contributor')).toBeVisible()
    })

    const row = getByRole('row', {
      name: /ariel\+01@storipress\.com/i,
    })

    const dropdownButton = within(row).getByRole('button', {
      name: /open-dropdown/i,
    })

    await fireEvent.click(dropdownButton)
    expect(within(row).getByRole('menu')).toBeVisible()
    expect(within(row).getByRole('menu')).toHaveTextContent('Revoke invitation')
    expect(within(row).getByRole('menu')).toHaveTextContent('Resend invitation')
  })
})

it('multiple select', async () => {
  const { getByText, getByRole } = render(Team)

  await waitFor(() => {
    expect(getByText('harvey contributor')).toBeVisible()
  })

  const ownerRow = getByRole('row', {
    name: /owner/i,
  })
  const invitedRow = getByRole('row', {
    name: /ariel\+01@storipress\.com/i,
  })
  const theadRow = getByRole('row', {
    name: /name status role desks/i,
  })

  expect(ownerRow).not.toContainHTML('checkbox')
  expect(invitedRow).not.toContainHTML('checkbox')

  await fireEvent.change(within(theadRow).getByRole('checkbox'), {
    target: { checked: true },
  })
  expect(getByText(/selected/i)).toHaveTextContent('9')
  expect(
    getByRole('button', {
      name: /suspend/i,
    }),
  ).toBeVisible()
  expect(
    getByRole('button', {
      name: /remove user/i,
    }),
  ).toBeVisible()
})
