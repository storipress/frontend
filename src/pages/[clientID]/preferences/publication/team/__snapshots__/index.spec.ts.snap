// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render settings team 1`] = `
[
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://assets.stori.press/storipress/storipress-helper-user-avatar.webp"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Storipress Helper
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://assets.stori.press/D6RX98VXN/avatars/01FZA2S6JVC6FPQY6X6FNR7TG6.jpg"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Michae Jackson
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/Test Testerson.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Test Testerson
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/Gary Chu.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Gary Chu
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/Google Pie.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Google Pie
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/David Peng.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      David Peng
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://assets.stori.press/D6RX98VXN/avatars/01FYE29VBFJ0Z4MWWTYC6NBAF0.gif"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Ariel Lin
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/Harvey Liu.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Harvey Liu
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/harvey author.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      harvey author
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/harvey contributor.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      harvey contributor
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="size-6 flex-none rounded-full border border-stone-900/10 bg-black/25"
    />
    <span
      class="ml-3 truncate text-stone-800"
    >
      ariel+01@storipress.com
    </span>
  </div>,
]
`;

exports[`sort settings team > sort by desks 1`] = `
[
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://assets.stori.press/D6RX98VXN/avatars/01FZA2S6JVC6FPQY6X6FNR7TG6.jpg"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Michae Jackson
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://assets.stori.press/D6RX98VXN/avatars/01FYE29VBFJ0Z4MWWTYC6NBAF0.gif"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Ariel Lin
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/harvey author.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      harvey author
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="size-6 flex-none rounded-full border border-stone-900/10 bg-black/25"
    />
    <span
      class="ml-3 truncate text-stone-800"
    >
      ariel+01@storipress.com
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/Test Testerson.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Test Testerson
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/Gary Chu.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Gary Chu
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/Google Pie.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Google Pie
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/harvey contributor.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      harvey contributor
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://assets.stori.press/storipress/storipress-helper-user-avatar.webp"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Storipress Helper
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/David Peng.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      David Peng
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/Harvey Liu.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Harvey Liu
    </span>
  </div>,
]
`;

exports[`sort settings team > sort by name 1`] = `
[
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://assets.stori.press/D6RX98VXN/avatars/01FYE29VBFJ0Z4MWWTYC6NBAF0.gif"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Ariel Lin
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="size-6 flex-none rounded-full border border-stone-900/10 bg-black/25"
    />
    <span
      class="ml-3 truncate text-stone-800"
    >
      ariel+01@storipress.com
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/David Peng.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      David Peng
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/Gary Chu.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Gary Chu
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/Google Pie.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Google Pie
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/harvey author.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      harvey author
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/harvey contributor.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      harvey contributor
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/Harvey Liu.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Harvey Liu
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://assets.stori.press/D6RX98VXN/avatars/01FZA2S6JVC6FPQY6X6FNR7TG6.jpg"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Michae Jackson
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://assets.stori.press/storipress/storipress-helper-user-avatar.webp"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Storipress Helper
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/Test Testerson.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Test Testerson
    </span>
  </div>,
]
`;

exports[`sort settings team > sort by role 1`] = `
[
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://assets.stori.press/D6RX98VXN/avatars/01FZA2S6JVC6FPQY6X6FNR7TG6.jpg"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Michae Jackson
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/Google Pie.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Google Pie
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/David Peng.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      David Peng
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://assets.stori.press/D6RX98VXN/avatars/01FYE29VBFJ0Z4MWWTYC6NBAF0.gif"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Ariel Lin
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/Harvey Liu.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Harvey Liu
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="size-6 flex-none rounded-full border border-stone-900/10 bg-black/25"
    />
    <span
      class="ml-3 truncate text-stone-800"
    >
      ariel+01@storipress.com
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/Test Testerson.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Test Testerson
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/harvey author.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      harvey author
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/Gary Chu.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Gary Chu
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/harvey contributor.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      harvey contributor
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://assets.stori.press/storipress/storipress-helper-user-avatar.webp"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Storipress Helper
    </span>
  </div>,
]
`;

exports[`sort settings team > sort by status 1`] = `
[
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://assets.stori.press/D6RX98VXN/avatars/01FZA2S6JVC6FPQY6X6FNR7TG6.jpg"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Michae Jackson
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/Test Testerson.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Test Testerson
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/Gary Chu.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Gary Chu
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/Google Pie.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Google Pie
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/David Peng.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      David Peng
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://assets.stori.press/D6RX98VXN/avatars/01FYE29VBFJ0Z4MWWTYC6NBAF0.gif"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Ariel Lin
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/Harvey Liu.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Harvey Liu
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/harvey author.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      harvey author
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://avatars.dicebear.com/api/croodles-neutral/harvey contributor.svg?b=%23F0F0F0"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      harvey contributor
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="size-6 flex-none rounded-full border border-stone-900/10 bg-black/25"
    />
    <span
      class="ml-3 truncate text-stone-800"
    >
      ariel+01@storipress.com
    </span>
  </div>,
  <div
    class="flex max-w-[10rem] items-center"
    data-testid="team-table-name-column"
  >
    <div
      class="flex flex-none items-center"
    >
      <div
        class="inline-block overflow-hidden rounded-full"
      >
        <img
          class="size-6"
          src="https://assets.stori.press/storipress/storipress-helper-user-avatar.webp"
        />
      </div>
    </div>
    <span
      class="ml-3 truncate text-stone-800"
    >
      Storipress Helper
    </span>
  </div>,
]
`;
